pipeline/config/config.go

```diff
@@ -43,10 +43,17 @@ type Pipeline struct {
 
 // Parameter defines a single parameter in a pipeline config
 type Parameter struct {
-	Name        string `yaml:"name"`
-	Description string `yaml:"description"`
-	Default     string `yaml:"default"`
-	Required    bool   `yaml:"required"`
+	Name        string   `yaml:"name"`
+	Description string   `yaml:"description"`
+	Default     string   `yaml:"default"`
+	Required    bool     `yaml:"required"`
+	HasOptions  bool     `yaml:"hasOptions"`
```

Comment:
do we need a bool `HasOptions` instead of just checking if the length of the `Options` slice is greater than 0? I see it's already in use sort of across tow different files so maybe it's too late now

Reply:
I can do that, sure. In the pipeline.yml file, that will just mean the `deployEmbeddedManifests` would require the dev to specify
```
deployEmbeddedManifests:
  ...
  options:
  - value: 'foo'
  - value: 'bar'
  ...
```

rather than
```
deployEmbeddedManifests:
  ...
  hasOptions: true
  options:
  - value: 'foo'
  - value: 'bar'
  ...


It seems as though the spinnaker json payload requires the bool and the list of options so I need to leave that part.